app/src/org/commcare/activities/QueryRequestUiController.kt

```diff
@@ -262,13 +262,11 @@ class QueryRequestUiController(
         promptSpinner.visibility = View.VISIBLE
         promptSpinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
             override fun onItemSelected(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {
-                var value = ""
                 if (position > 0) {
                     val choices = queryPrompt.itemsetBinding!!.choices
                     val selectChoice = choices[position - 1]
-                    value = selectChoice.value
+                    updateAnswerAndRefresh(queryPrompt, selectChoice.value)
```

Comment:
@avazirna what prompted this change ? What if user clears the spinner choice and selects nothing ?

Reply:
@shubham1g5 because of an infinite loop, when the data source is set for the first time and `onItemSelected` is called, it calls the [updateAnswerAndRefresh](https://github.com/dimagi/commcare-android/blob/53398c06f6bf08b9543c90a3c92f2d9f517195f0/app/src/org/commcare/activities/QueryRequestUiController.kt#L197C17-L197C39) with an empty `String` value, and in that method when `oldAnswer == null || !oldAnswer.contentEquals(answer)` is true, we are now setting the user answer value to `null` which means that when we call [refreshView()](https://github.com/dimagi/commcare-android/blob/53398c06f6bf08b9543c90a3c92f2d9f517195f0/app/src/org/commcare/activities/QueryRequestUiController.kt#L61C24-L61C31) and refresh the data source again, it triggers `onItemSelected` with an empty String,  `oldAnswer == null || !oldAnswer.contentEquals(answer)` evaluates to true again, and so on. But you are right, this option misses the scenario when the user changes back to an, I thought the index started form 1. Let me fix that, there is another scenario I tested with.